app/src/main/java/com/hover/stax/transfers/NonTemplateVariable.kt

```diff
@@ -0,0 +1,19 @@
+package com.hover.stax.transfers
+
+import java.util.*
+
+data class NonTemplateVariable(val key: String,
```

Comment:
The fact this is empty is a code smell. Objects should have a purpose, and if they contain nothing that means either code that is their responsibility is being done elsewhere or they should not exist. In this case I think it is that it should not exist. 

The view state does not need to be stored: since we aren't storing the view state for the rest of the form this leads to different behavior from these fields than the other fields. View state can be recalculated if necessary. A LinkedHashMap can be used to store the variables in the view model which will actually simplify the logic in a number of places